proposals/NNNN-unlock-existentials-with-fixed-associated-types.md

```diff
@@ -0,0 +1,249 @@
+
+To reflect these new guidelines the 7th rule for 'allowed' changes ([protocol section](https://github.com/apple/swift/blob/master/docs/LibraryEvolution.rst#protocols)) will be removed. The rule to be changed, states that: 
+> A new associatedtype requirement may be added (with the appropriate availability), as long as it has a default implementation.
+
+This rule will be replaced by the following rule and be moved into the 'forbidden' section:
+> Adding `associatedtype` requirements (even if they have a default implementation) 
+
+
+## Alternatives considered
+
+### Do Nothing
+
+The current design is quite problematic - as discussed in the [Motivation](#motivation) section. Not to mention, it seems like an abnormality in the generics and exitentials system. There has, also, been [post](https://forums.swift.org/t/making-a-protocols-associated-type-concrete-via-inheritance/6557) after [post](https://forums.swift.org/t/constrained-associated-type-on-protocol/38770) asking why this feature isn’t yet implemented. Fixing this ‘issue’ will strengthen the foundation of the existentials systems to allow for [more and exciting future additions](https://forums.swift.org/t/improving-the-ui-of-generics/22814)
+
+
+### Disallow Constraining an Associated Type to the Protocol's Existential
+
+As mentioned in the [What would be a Regular Protocol](#what-would-be-a-regular-protocol?) section, the sixth example demonstrates how a pretty odd case would be handled. If you think about it, though, it’s not that different from constraining an associated type to its enclosing protocol - such as SwiftUI’s [`View`](https://developer.apple.com/documentation/swiftui/view). Furthermore, protocols such as the following one are currently allowed: 
+```swift
+protocol Foo {
+    var foo: Foo { get }
+}
+```
+Moreover, we are not actually constraining `Foo`'s `A` to `Foo` itself but rather its _Existential_. Not to mention, that in the future all protocols might be allowed to have Existentials (read more in the [Generalized Existentials](#generalized-existentials) section).
+
+All in all, we don’t think it’s for the compiler to warn us when a protocol is _likely_ to fail, but rather when failure is _certain_, due to protocols' abstract nature.
+
+## Future Directions
+
+### Generalized Existentials 
+
+#### What are They Exactly? 
+
+An Existential of a Protocol is a type - separate from the protocol itself - that can contain any value of a type that conforms to a given set of constraints - which include conforming to that protocol. Such types are currently provided for regular protocols under the same name, as seen in this example:
```

Comment:
Could you elaborate on why? I think an introduction is apt for such a complex topic. Do you maybe have some different kind of introduction in mind?

Reply:
This proposal is simply lifting some restrictions on a *long-standing* feature. We can add links, including to [Improving the UI of generics](https://forums.swift.org/t/improving-the-ui-of-generics/22814), for anyone having trouble understanding the concepts behind the proposal, but I would rather not reinvent a tutorial on or introduction to something that can already be looked up in the forums or Swift book. It would also take time to review and will probably end up distracting the reader.